core/feconf.py

```diff
@@ -24,11 +24,28 @@
 
 from core.constants import constants
 
-from typing import Dict, List, Union
+from typing import Dict, List
 from typing_extensions import TypedDict
 
-CommandType = (
-    Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]])
+
+# This TypedDict is used only to create the TypedDict below, because we want
+# to mark some attributes as optional.
+class _ValidCmdDictNotTotal(TypedDict, total=False):
+    """Dictionary representing a non-required part
+    of valid commands specs."""
+
+    allowed_values: Dict[str, List[str]]
+    deprecated_values: Dict[str, List[str]]
+
+
+class ValidCmdDict(_ValidCmdDictNotTotal, total=False):
```

Comment:
Same thing, do you need total=False here? Ideally everything has a consistent format.

Reply:
In this scenario, total = False is used as `allowed_values` and `deprecated_values` were non-required and we wanted to inherit it to ValidCmdDict, to make the latter total.

Reply:
Can you now remove the `total=False`?

Reply:
If I remove `total=False` now, then I am getting the following single error

![image](https://user-images.githubusercontent.com/65958420/155267922-31941bb2-7161-4acf-96a8-eee27ec08609.png)

I think is because, in change_domain.py, `valid_cmd_attribute_specs`, is set as `None`, so if in some case it has no value of `name` in it, and when the dict is total, it would require the `name` key everywhere.

Reply:
No, it is because we `pop` some key-values from the dict in _change_domain.py_, can you try to refactor the code there to not `pop` the stuff?

Reply:
@div-yam Any reply here?

Reply:
The types in keys must be `Set[Literal['name']]`